ideascube/serveradmin/catalog.py

```diff
@@ -176,6 +185,97 @@ def unregister_site(self, package):
+        #ensure that mandatory fields are set
+        for entry in ('title', 'summary', 'path', 'credits'):
+            if entry not in media_info:
+                raise InvalidPackageContent('Missing {} in {}'.format(entry, media_info))
+
+        files = {'original': None, 'preview': None}
+
+        title = smart_truncate(media_info['title'])
+
+        lang = media_info.get('lang', settings.LANGUAGE_CODE)
+
+        kind = media_info.get('kind')
+        if not kind or not hasattr(Document, kind.upper()):
+            content_type, _ = mimetypes.guess_type(original)
+            kind = guess_kind_from_content_type(content_type) or Document.OTHER
+
+        instance = Document.objects.filter(title=title, kind=kind).last()
+        if instance:
+            raise InvalidPackageContent('Document {} already exists'.format(title))
+
+        metadata = {
+            'title'      : title,
+            'summary'    : media_info['summary'],
+            'credits'    : media_info['credits'],
+            'tags'       : media_info.get('tags'),
+            'lang'       : lang,
+            'kind'       : kind,
+            'package_id' : package_id
+        }
+
+        o, p = None, None
+        try:
+            original = media_info['path']
+            try:
+                o = Path(install_dir, original).open('rb')
+                files['original'] = File(o, name=original)
+            except OSError as e:
+                raise InvalidPackageContent('Cannot open path {}. Error is {}'.format(path, e))
+
+            if 'preview' in media_info:
+                preview = media_info['preview']
+                try:
+                    p = Path(install_dir, preview).open('rb')
+                    files['preview'] = File(p, name=preview)
+                except OSError:
+                    pass
+
+            self._save_media(metadata, files)
+        finally:
+            # close opened file whatever happen
+            [f.close() for f in (o, p) if f]
+
+    def _save_media(self, metadata, files):
+        form = DocumentForm(data=metadata, files=files, instance=None)
+
+        if form.is_valid():
+            doc = form.save()
+            print("Media", doc, "has been added to database.")
+        else:
+            lerr = ["Some values are not valid :"]
+            for field, error in form.errors.items():
+                lerr.append(" - {}: {}".format(field, error.as_text()))
+            raise InvalidPackageContent("\n".join(lerr))
+
+    def install_pkg(self, pkg):
+        print('Adding medias to mediacenter database.')
+        root = pkg.get_root_dir(self._install_dir)
+        manifestfile = Path(root, 'manifest.yml')
+        with manifestfile.open('r') as m:
+            manifest = yaml.safe_load(m.read())
+
+        for media in manifest['medias']:
+            try:
+                self._install_media(media, root, pkg.id)
+            except:
+                # What to do here ? revert/stop/continue ?
```

Comment:
At least print a warning.
This could lead to package installed but not all medias accessible from mediacenter in case of errors.